### PR TITLE
Added Diamond Square Code

### PR DESCRIPTION
**Pull request by BirajCoder**
# Added Diamond Sqaure Code 

Added the diamond square Code in Python

Related: Issue #22

## Type of change

- [ Y] New feature (non-breaking change which adds functionality)

# Checklist:

- [ Y] My code follows the style guidelines of this project
- [ Y] I have performed a self-review of my own code
- [ Y] I have commented my code, particularly in hard-to-understand areas
- [ Y] I have made corresponding changes to the documentation
- [ Y] My changes generates no new warnings
- [ Y] I have added tests that prove my fix is effective or that my feature works
- [ Y] New and existing unit tests pass locally with my changes
- [ Y] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: snippets/python/Diamond_Square.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Diamond Square
+
+## Description
+A Program to print the Diamond Sqaure Pattern, number of rows is given as input.
+
+## Code
+```python
+# The Diamond Pattern
```
Comment (kaustubhgupta) [inline snippets/python/Diamond_Square.md]: This comment is not appropriate for the code you are referring to. As I can see you are taking the number of rows as input, you can simply quote it as 'rows input'

File: snippets/python/Diamond_Square.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+n=int(input("Enter number of Rows: "))
+
+# The Diamond Pattern
```
Comment (kaustubhgupta) [inline snippets/python/Diamond_Square.md]: Same applies here

File: snippets/python/Diamond_Square.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    print(space+star)
+```
+
+## Input
+```
+Enter number of Rows: 5
+```
+
+## Output
+```
```
Comment (kaustubhgupta) [inline snippets/python/Diamond_Square.md]: Put the image of the output of the console instead of hardcoding the output

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: snippets/python/Diamond_Square.md (hunk 1 above)
Comment (BirajCoder) [reply snippets/python/Diamond_Square.md]: Thanks , for the review , will upload the new file soon.

File: snippets/python/Diamond_Square.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+# The Top Pyramid
+for i in range(n):
+    space="  "*(n-i-1)
+    star=(2*i)*'* '+'*'
+    print(space+star)
+    
+# The bottom Pyramid
+for i in range(n):
+    space = ("    " if i!=0 else "  ")+"  "*(i-1)
```
Comment (sakshamarora1) [inline snippets/python/Diamond_Square.md]: The code looks a bit messy. You can reformat the code with some formatter like black, etc. That will make the code more readable!

Comment (BirajCoder) [reply snippets/python/Diamond_Square.md]: I was unaware of code formatter , thanks for the suggestion.

File: snippets/python/Diamond_Square.md (hunk 3 above)
Comment (BirajCoder) [reply snippets/python/Diamond_Square.md]: Hey, to upload the image in the markdown file I need to put the image in markdown file too , shall I make another directory inside the python directory named as **Output Images** . Please tell me if there is any other way to include an image in the markdown file.

Comment (kaustubhgupta) [reply snippets/python/Diamond_Square.md]: You can create another folder under the Python folder named "Diamond Pattern"

**Comment by BirajCoder** [comment]
Hey I have made all the required modifications in my code and also used a formatter and I have updated my descriptions. Please review my modifications.

File: snippets/python/Diamond_Square/Diamond_Square.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Diamond Square
+
+## Description
+A Program to print the Diamond Sqaure Pattern. The rows count is given as a input which denotes the longest row in the pattern.
+
+## Code
+```python
+# The Diamond Pattern
```
Comment (sakshamarora1) [inline snippets/python/Diamond_Square/Diamond_Square.md]: Redundant comment
```suggestion
```

File: snippets/python/Diamond_Square/Diamond_Square.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    space = "  " * (i)
+    star = star[:-4]
+    print("  " + space + star)
+```
+
+## Input
+```
+Enter the rows count: 7
+```
+
```
Comment (sakshamarora1) [inline snippets/python/Diamond_Square/Diamond_Square.md]: The input line will already be visible in the output image. So there's no need for an extra sub heading.
```suggestion
```

File: snippets/python/Diamond_Square/Diamond_Square.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+## Output
+
+![Code Output](./Output.png)
```
Comment (sakshamarora1) [inline snippets/python/Diamond_Square/Diamond_Square.md:25]: The image is not correctly cropped. Remove this and add a perfectly cropped image. Also keep the input line here itself.

**Review by sakshamarora1 — CHANGES_REQUESTED** [review]
Just a few changes!

File: snippets/python/Diamond_Square/Diamond_Square.md (hunk 7 above)
Comment (BirajCoder) [reply snippets/python/Diamond_Square/Diamond_Square.md:25]: Ok sure , Thank you

**Comment by BirajCoder** [comment]
Made the necessary modifications , let me know if there are any more improvements required in this code,

**Review by sakshamarora1 — APPROVED** [review]
Looks good to me!

**Review by kaustubhgupta — APPROVED** [review]

**Pull request merged**